lib/src/main/java/com/iwillfailyou/javaparser/NodeCaller.java

```diff
@@ -0,0 +1,26 @@
+package com.iwillfailyou.javaparser;
+
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.body.MethodDeclaration;
+import com.github.javaparser.ast.expr.LambdaExpr;
+import javax.annotation.Nonnull;
+
+public final class NodeCaller implements Caller {
+
+    final Node callee;
+
+    public NodeCaller(@Nonnull final Node callee) {
+        this.callee = callee;
+    }
+
+
+    @Override
+    public Node caller() {
+        return callee.getParentNode()
```

Comment:
initially, I created this as a private static method, but then I wasn't sure on your opinion on private static methods. I saw you opened https://github.com/iwillfailyou/java-inspections/issues/23, but still not sure you want to allow private static in your codebase.

Reply:
Yeah, any private method is a reason for a new class